File: app/src/main/java/com/neocaptainnemo/testing/ui/NavigationDecisionDialogFragment.java
package com.neocaptainnemo.testing.ui;

import android.app.Dialog;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.neocaptainnemo.testing.R;
import com.neocaptainnemo.testing.model.AtmNode;

import java.util.Formatter;
import java.util.Locale;

public class NavigationDecisionDialogFragment extends DialogFragment {

    static final String TAG = "NavigationDecisionDialogFragment";

    static NavigationDecisionDialogFragment instance(@NonNull AtmNode atmNode) {
        NavigationDecisionDialogFragment fragment = new NavigationDecisionDialogFragment();
        Bundle arg = new Bundle();
        arg.putParcelable("atm", atmNode);
        fragment.setArguments(arg);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.choose_the_nav_way)
                .setItems(R.array.nav_way_array, (dialog, which) -> {

                    AtmNode atm = getArguments().getParcelable("atm");

                    StringBuilder geoBuilder = new StringBuilder();
                    Formatter formatter = new Formatter(geoBuilder, Locale.US);

                    switch (which) {
                        case 0:
                            formatter.format("google.navigation:q=%f,%f&mode=%s", atm.getLat(), atm.getLon(), "w");
                            break;

                        case 1:
                            formatter.format("google.navigation:q=%f,%f&mode=%s", atm.getLat(), atm.getLon(), "d");
                            break;

                        default:
                            //do nothing
                            break;
                    }


                    Uri gmmIntentUri = Uri.parse(geoBuilder.toString());
                    Intent mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
                    startActivity(mapIntent);

                });
        return builder.create();
    }
}

File: app/src/main/java/com/neocaptainnemo/atms/ui/NavigationDecisionDialogFragment.kt
package com.neocaptainnemo.atms.ui

import android.app.Dialog
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import com.neocaptainnemo.atms.R
import com.neocaptainnemo.atms.model.AtmNode
import java.util.*

class NavigationDecisionDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!)
        builder.setTitle(R.string.choose_the_nav_way)
                .setItems(R.array.nav_way_array) { _, which ->

                    val atm = arguments!!.getParcelable<AtmNode>("atm")

                    val geoBuilder = StringBuilder()
                    val formatter = Formatter(geoBuilder, Locale.US)

                    when (which) {
                        0 -> formatter.format("google.navigation:q=%f,%f&mode=%s", atm!!.lat, atm.lon, "w")

                        1 -> formatter.format("google.navigation:q=%f,%f&mode=%s", atm!!.lat, atm.lon, "d")

                        else -> {
                        }
                    }//do nothing


                    val gmmIntentUri = Uri.parse(geoBuilder.toString())
                    val mapIntent = Intent(Intent.ACTION_VIEW, gmmIntentUri)
                    startActivity(mapIntent)

                }
        return builder.create()
    }

    companion object {

        const val tag = "NavigationDecisionDialogFragment"

        internal fun instance(atmNode: AtmNode): NavigationDecisionDialogFragment {
            val fragment = NavigationDecisionDialogFragment()
            val arg = Bundle()
            arg.putParcelable("atm", atmNode)
            fragment.arguments = arg
            return fragment
        }
    }
}
